common.rkt

```diff
@@ -146,13 +146,13 @@
                                (state (extend-sub t (reified-index index) (state-sub st)) (state-diseq st) (state-types st)))
                     (else      st)))))
     (let* ((walked-sub (walk* tm results))
-          (diseq-null (null? (state-diseq st)))
-          (types-null (null? (state-types st)))
-          (types (walk* (map pretty-types (state-types st)) results))
-          (diseq (if diseq-null '() (list (walk* (cons '=/= (map pretty-diseq (state-diseq st))) results)))))
-      (if (and diseq-null types-null)
+           (types (walk* (map pretty-types (state-types st)) results))
+           (diseq (map pretty-diseq (state-diseq st)))
+           (diseq (if (null? diseq) '() (list (walk* (cons '=/= diseq) results))))
+           (cxs (append types diseq)))
+      (if (null? cxs)
           walked-sub
-          (Ans walked-sub (append types diseq))))))
+          (Ans walked-sub cxs)))))
```

Comment:
The final improvement we should make to `reify` is to normalize the order of `cxs`.  You can do this by defining a `term<=?` and `sort`ing `cxs` with it.

It doesn't matter which order we use, as long as it's consistent.  Though it might be nice to match the order given in other miniKanren implementations.